src/build-application.js

```diff
@@ -26,18 +26,20 @@ function bundleNode(inputFolder, outputFolder, watch) {
 
     return new Promise((resolve, reject) => {
       const inputFilename = pathname;
-      const outputFilename = inputFilename.replace(inputFolder, outputFolder);
+      // TODO: Check if .tsx support is working as expected
```

Comment:
Here are the the Babel config adjustments: When using the `transformFile` call, like you did, Babel is basically making the TS->JS conversion, but the files would be still written to the .build folder with the `.ts` extension. Therefore, I've replaced the file path by hand. There maybe is a better solution out there for these cases, but I hadn't find one.  

To make sure and notify the user of the renaming, I've added the additional console output if the transpile step is executed with a renaming action.